Packs/ApiModules/Scripts/JSONFeedApiModule/JSONFeedApiModule.py

```diff
@@ -210,32 +229,31 @@ def extract(json_element):
 
 def feed_main(params, feed_name, prefix):
     handle_proxy()
-
     client = Client(**params)
     indicator_type = params.get('indicator_type')
     feedTags = argToList(params.get('feedTags'))
+    limit = int(demisto.args().get('limit', 10))
     command = demisto.command()
     if prefix and not prefix.endswith('-'):
         prefix += '-'
     if command != 'fetch-indicators':
         demisto.info(f'Command being called is {demisto.command()}')
     try:
         if command == 'test-module':
-            return_outputs(test_module(client, params))
+            return_results(test_module(client, limit))
 
         elif command == 'fetch-indicators':
-            indicators = fetch_indicators_command(client, params.get('indicator_type'), feedTags,
+            indicators = fetch_indicators_command(client, indicator_type, feedTags,
                                                   params.get('auto_detect_type'))
             for b in batch(indicators, batch_size=2000):
                 demisto.createIndicators(b)
 
         elif command == f'{prefix}get-indicators':
             # dummy command for testing
-            limit = int(demisto.args().get('limit', 10))
             auto_detect = params.get('auto_detect_type')
-            indicators = fetch_indicators_command(client, indicator_type, feedTags, auto_detect)[:limit]
+            indicators = fetch_indicators_command(client, indicator_type, feedTags, auto_detect, limit)
```

Comment:
let's call `auto_detect` where we call `indicator_type` so we'll do at l246: 
fetch_indicators_command(client, indicator_type, feedTags, auto_detect) as well
and it will look nicer.